### PR TITLE
Module 3.1

### PR DESCRIPTION
**Pull request by AuTiiTo**

### CODE REVIEW

File: app/src/main/java/sergio/splashbaseviewer/mvp/model/SplashModel.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            }
+
+            @Override
+            public void onFailure(Call<SplashBaseResponse> call, Throwable t) {
+                consumer.onFailure(t.getMessage());
+            }
+        });
+    }
+
+    public interface SplashServiceConsumer {
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/mvp/model/SplashModel.java]: Remove this interface. You should use otto event.

File: app/src/main/java/sergio/splashbaseviewer/mvp/model/SplashModel.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+public class SplashModel {
+
+    private final SplashServiceConsumer consumer;
+
+    public SplashModel(SplashServiceConsumer consumer) {
+        this.consumer = consumer;
+    }
+
+    public void startDownload() {
+        SplashBaseService service = SplashBaseService.retrofit.create(SplashBaseService.class);
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/mvp/model/SplashModel.java]: You should separate the services functionality in another class. This another class is invoked for the model.

File: app/src/main/java/sergio/splashbaseviewer/mvp/presenter/SplashPresenter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+public class SplashPresenter implements SplashModel.SplashServiceConsumer{
+    @NonNull
+    private SplashView view;
+    @NonNull
+    private SplashModel model;
+
+    public SplashPresenter(SplashView view) {
+        this.model = new SplashModel(this);
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/mvp/presenter/SplashPresenter.java]: The objects model and view are Instantiated in the Activity. Never in the presenter. Remmember this. :)

File: app/src/main/java/sergio/splashbaseviewer/mvp/view/SplashView.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        super(mainActivity);
+        this.bus = bus;
+        ButterKnife.bind(this, mainActivity);
+    }
+
+    public void setResul(String result) {
+        jsonMain.setText(result);
+    }
+
+    @OnClick(R.id.fab_splash_sync)
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/mvp/view/SplashView.java]: The FAB button is to point 4 of module 3.

File: app/src/main/java/sergio/splashbaseviewer/utils/SplashBaseService.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.http.GET;
+import sergio.splashbaseviewer.utils.models.SplashBaseResponse;
+
+/**
+ * @author s.ruiz
+ */
+
+public interface SplashBaseService {
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/utils/SplashBaseService.java:11]: Separate this please. Create an interface for declaration of calls, and create a class to manage the calls.

File: app/src/main/java/sergio/splashbaseviewer/utils/models/SplashBaseResponse.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package sergio.splashbaseviewer.utils.models;
+
+import java.util.List;
+
+/**
+ * @author s.ruiz
+ */
+public class SplashBaseResponse {
+    private List<SplashImage> images;
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/utils/models/SplashBaseResponse.java]: As best practice, the class SplashImage should be named SplashImageResponse.

**Comment by nacho-globant** [comment]
Code review with comments.

File: app/src/main/java/sergio/splashbaseviewer/utils/SplashBaseService.java (hunk 5 above)
Comment (AuTiiTo) [reply app/src/main/java/sergio/splashbaseviewer/utils/SplashBaseService.java:11]: Ready

File: app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFailureEvent.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package sergio.splashbaseviewer.mvp.events;
+
+/**
+ * @author s.ruiz
+ */
+
+public class DownloadFailureEvent {
+    String errorMessage;
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFailureEvent.java:8]: Dont modify this. But if you define the getter method, the field should be private.

File: app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFinishedEvent.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.List;
+
+import sergio.splashbaseviewer.utils.models.SplashImageResponse;
+
+/**
+ * @author s.ruiz
+ */
+
+public class DownloadFinishedEvent {
+    List<SplashImageResponse> images;
```
Comment (nacho-globant) [inline app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFinishedEvent.java:12]: Dont modify this. But if you define the getter method, the field should be private.

**Comment by nacho-globant** [comment]
+1 with some observations. Modify these in the next PR. 👍

File: app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFailureEvent.java (hunk 7 above)
Comment (AuTiiTo) [reply app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFailureEvent.java:8]: OK

File: app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFinishedEvent.java (hunk 8 above)
Comment (AuTiiTo) [reply app/src/main/java/sergio/splashbaseviewer/mvp/events/DownloadFinishedEvent.java:12]: OK

**Pull request merged**